corehq/apps/hqwebapp/management/commands/build_requirejs.py

```diff
@@ -37,251 +35,261 @@ def add_arguments(self, parser):
+                            "--noinput && ./manage.py compilejsi18n` (y/n)? ")
+            if confirm[0].lower() != 'y':
+                exit()
 
         # During deploy, resource_static should already have run and populated resource_versions
         from get_resource_versions import get_resource_versions
-        resource_versions = get_resource_versions()
-        if (not resource_versions):
+        self.resource_versions = get_resource_versions()
+        if (not self.resource_versions):
             raise ResourceVersionsNotFoundException()
 
-        config, local_js_dirs = _r_js(local=local, verbose=verbose, bootstrap_version=bootstrap_version)
-        if optimize:
-            _minify(config, verbose=verbose)
+    def _staticfiles_path(self, *parts):
+        return os.path.join(settings.BASE_DIR, 'staticfiles', *parts)
+
+    def _get_html_files_and_local_js_dirs(self):
+        """
+        Returns
+        - all HTML files in corehq, excluding partials
+        - a reference of js directories, for use when copying optimized bundles back into corehq
+        """
+        logger.info("Gathering HTML templates and JS directories")
+        html_files = []
+        local_js_dirs = set()
+        for root, dirs, files in os.walk(os.path.join(settings.BASE_DIR, 'corehq')):
+            for name in files:
+                if name.endswith(".html"):
+                    filename = os.path.join(root, name)
+                    if '/partials/' not in filename:
+                        if '/includes/' not in filename and '/_includes/' not in filename:
+                            html_files.append(filename)
+                elif self.local and name.endswith(".js"):
+                    local_js_dirs.add(os.path.relpath(root))
+        return html_files, local_js_dirs
+
+    def _get_main_js_modules_by_dir(self, html_files):
+        """
+        Returns a dict of all main js modules, grouped by bootstrap version and then by directory:
+        {
+            3: {
+                'linked_domain/js': set(['linked_domain/js/domain_links']),
+                'locations/js': set(['locations/js/import', ...  ]),
+                ...
+            },
+            5: {
+                'data_dictionary/js': set(['data_dictionary/js/data_dictionary']),
+                'locations/js': set(['locations/js/location', ...  ]),
+                ...
+            }
+        }
+        """
+        logger.info("Mapping main modules to django apps")
+        dirs = {v: defaultdict(set) for v in BOOTSTRAP_VERSIONS}
+        for filename in html_files:
+            proc = subprocess.Popen(["grep", r"^\s*{% requirejs_main [^%]* %}\s*$", filename],
+                                    stdout=subprocess.PIPE)
+            (out, err) = proc.communicate()
+            out = out.decode('utf-8')
+            if out:
+                match = re.search(r"{% requirejs_main(_b([35]))? .(([^%]*)/[^/%]*). %}", out)
```

Comment:
This would only ever match `requirejs_main_b5` since `requirejs_main_b3` does not exist, right?

It would make this file easier to update in the future if this and the logic below used values from `BOOTSTRAP_VERSIONS` rather than having them hard-coded.

Reply:
yes, I must have written this while planning to introduce both `requirejs_main_b3` and `requirejs_main_b5` and then decided not to do that. So I could hard-code the `5` in the regex. That's less flexible regarding future changes, but realistically, I'm anticipating removing all version-related logic once the bootstrap migration is done.

Reply:
I ultimately updated this to only reference b5, in https://github.com/dimagi/commcare-hq/pull/34346/commits/4048d572c890b63ca22dc84c55eca2c017a7e2da